kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -4789,6 +4795,43 @@ def hotplug_cpu(self, cpu_num):
             raise kvmagent.KvmError(err)
         return
 
+    def _qga_online_hotplugged_cpus(self, prev_cpu_num):
+        """After CPU hotplug, use QGA guest-set-vcpus to online only the newly added CPUs.
+
+        @param prev_cpu_num: the CPU count before this hotplug operation,
+               used to avoid re-onlining CPUs that were manually offlined by the user.
+        """
+        try:
+            qga = VmQga(self.domain)
```

Ruff (0.15.7)

[error] `VmQga` may be undefined, or defined from star imports

(F405)